nexus/operation.go

```diff
@@ -36,6 +38,13 @@ func (r operationReference[I, O]) Name() string {
 	return string(r)
 }
 
+func (r operationReference[I, O]) IsInputAssignable(i any) bool {
```

Comment:
Not sure this needs to be on every operation reference (nor does `IsOutputAssignable`). If you had a `InputType() reflect.Type` that could maybe make a bit more sense, but anyone can obtain the generic of the operation reference via reflection. IMO having a helper for Go reflection is unnecessary for this public API

Can just update the Temporal side to use reflection if it needs to check whether something is assignable, same as it would for any generic API it would need to do a similar check for.

Reply:
I don't think it's possible to get the generic input type in Go. I tried this before and ended up blocked on https://github.com/golang/go/issues/54393.

Maybe there's another way you're aware of that I'm not.

Reply:
I see, my mistake. I think `InputType() reflect.Type` and `OutputType() reflect.Type` would be better general-purpose methods to add.

Reply:
That works for me.